libbeat/outputs/elasticsearch/client.go

```diff
@@ -374,29 +385,23 @@ func getPipeline(data outputs.Data, pipelineSel *outil.Selector) (string, error)
 	}
 
 	if pipelineSel != nil {
-		return pipelineSel.Select(data.Event)
+		return pipelineSel.Select(event)
 	}
 	return "", nil
 }
 
 // getIndex returns the full index name
 // Index is either defined in the config as part of the output
 // or can be overload by the event through setting index
-func getIndex(event common.MapStr, index outil.Selector) string {
-
-	ts := time.Time(event["@timestamp"].(common.Time)).UTC()
-
-	// Check for dynamic index
-	// XXX: is this used/needed?
-	if _, ok := event["beat"]; ok {
-		beatMeta, ok := event["beat"].(common.MapStr)
-		if ok {
-			// Check if index is set dynamically
-			if dynamicIndex, ok := beatMeta["index"]; ok {
-				if dynamicIndexValue, ok := dynamicIndex.(string); ok {
-					return fmt.Sprintf("%s-%d.%02d.%02d",
-						dynamicIndexValue, ts.Year(), ts.Month(), ts.Day())
-				}
+func getIndex(event *beat.Event, index outil.Selector) string {
+
+	if event.Meta != nil {
+		if str, exists := event.Meta["index"]; exists {
+			idx, ok := str.(string)
+			if ok {
```

Comment:
Which method for setting your own index?
We used to have some logic to first look into the event, then look into the event metadata and then use the actual outputs index setting.

Was the `first` method ever used? I did keep the logic of allowing a beat to set index and ingest pipeline via event metadata fields (used/configured by filebeat modules).